helm/grafana/templates/grafana-deployment.yaml

```diff
@@ -23,6 +23,8 @@ spec:
       - name: grafana
         image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
         env:
+        - name: GF_SERVER_ROOT_URL
+          value: "{{ .Values.server_root_url }}"
         - name: GF_AUTH_BASIC_ENABLED
```

Comment:
to avoid the ingress fqdn confict with server_root_url, it's better check if .Values.ingress.enabled, if the .Values.ingress.enabled then we should ignore the server_root_url, if .Values.ingress.enabled is false, we can set .Values.server_root_url.